daemon/cluster/cluster.go

```diff
@@ -1070,23 +1070,40 @@ func (c *Cluster) ServiceLogs(ctx context.Context, input string, config *backend
 		return c.errNoManager(state)
 	}
 
+	var stream swarmapi.Logs_SubscribeLogsClient
 	service, err := getService(ctx, state.controlClient, input)
 	if err != nil {
-		c.mu.RUnlock()
-		return err
-	}
-
-	stream, err := state.logsClient.SubscribeLogs(ctx, &swarmapi.SubscribeLogsRequest{
-		Selector: &swarmapi.LogSelector{
-			ServiceIDs: []string{service.ID},
-		},
-		Options: &swarmapi.LogSubscriptionOptions{
-			Follow: config.Follow,
-		},
-	})
-	if err != nil {
-		c.mu.RUnlock()
-		return err
+		task, err := getTask(ctx, state.controlClient, input)
+		if err != nil {
+			c.mu.RUnlock()
+			logrus.Debug(err)
+			return fmt.Errorf("neither a service nor a task: %s", input)
```

Comment:
Should this be `no such service or task: %q`?

Also wondering if `errors.Wrapf()` should be used instead of the `Debug()`, but not sure if it adds much useful information